extensions/info/deployment/src/main/java/io/quarkus/info/deployment/InfoProcessor.java

```diff
@@ -0,0 +1,129 @@
+package io.quarkus.info.deployment;
+
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.PersonIdent;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.lib.RepositoryBuilder;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.jboss.logging.Logger;
+
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.bootstrap.workspace.WorkspaceModule;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+import io.quarkus.info.deployment.spi.InfoBuildTimeContributorBuildItem;
+import io.quarkus.info.deployment.spi.InfoBuildTimeValuesBuildItem;
+import io.quarkus.info.runtime.InfoRecorder;
+import io.quarkus.info.runtime.spi.InfoContributor;
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.vertx.http.deployment.NonApplicationRootPathBuildItem;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+
+public class InfoProcessor {
+
+    private static final Logger log = Logger.getLogger(InfoProcessor.class);
+
+    @BuildStep(onlyIf = GitInInfoEndpointEnabled.class)
+    InfoBuildTimeValuesBuildItem gitInfo(CurateOutcomeBuildItem curateOutcomeBuildItem) {
+        ApplicationModel applicationModel = curateOutcomeBuildItem.getApplicationModel();
+        WorkspaceModule workspaceModule = applicationModel.getAppArtifact().getWorkspaceModule();
+        if (workspaceModule == null) {
```

Comment:
Just in case, this would happen in a prod build. A workspace would be available only in dev and test modes (unless it's a platform testsuite).

Reply:
We could probably enable workspace discovery in prod build mode too but still resolve local dependencies as JARs instead of classes dirs.

Reply:
I don't think it's terribly important TBH, WDYT?

Reply:
Is this meant to work only in dev mode?

Reply:
No, it's meant to create data for the `info` endpoint which is available in all modes. 

When I did `mvn clean package -DskipTests`, the `WorkspaceModule` was properly properly populated, so I feel I don't understand something of what you are proposing :)

Reply:
That shouldn't happen in a prod build unless `quarkus.bootstrap.workspace-discovery` system property is true.

Reply:
Or a project property.

Reply:
I don't see it being set, but I'll have a look soon.

Reply:
It seems like workspace discovery is inadvertently  performed [here](https://github.com/quarkusio/quarkus/blob/aa3be9ef39af4daf3a0558f760e09ce55dbc8a6e/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java#L226) while the configuration you mention is not taken into account until [here](https://github.com/quarkusio/quarkus/blob/aa3be9ef39af4daf3a0558f760e09ce55dbc8a6e/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java#L253).

I'll make the git handling code more resilient to take into account how workspace discovery is supposed to work.

Reply:
The workspace discovery is performed https://github.com/quarkusio/quarkus/blob/aa3be9ef39af4daf3a0558f760e09ce55dbc8a6e/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java#L172
Ok, it's kind of "broken" due to https://github.com/quarkusio/quarkus/blob/aa3be9ef39af4daf3a0558f760e09ce55dbc8a6e/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java#L170
If it's not disabled, it's on by default.

Reply:
👍🏼 I'll leave it to you to determine what the proper behavior is.

In the meantime, I'll make the new code work whether or not the workspace is around.

Reply:
I have updated the PR and also added an advanced test to make sure that the Git handling does not blow up when there is no Git repo.